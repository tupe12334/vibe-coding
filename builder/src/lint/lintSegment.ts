import { readFile } from "fs/promises";
import { join } from "path";
import type { ListItem, RootContent } from "mdast";

export const lintSegment = async (
  templatesPath: string,
  lintSystem: string
): Promise<RootContent[]> => {
  const lintJsonFile = (
    await readFile(join(templatesPath, "lint", `${lintSystem}.json`), {
      encoding: "utf-8",
    })
  ).toString();
  const lintItems = JSON.parse(lintJsonFile);

  const segment: RootContent[] = [
    {
      type: "heading",
      depth: 2,
      children: [
        { type: "text", value: "Lint system" },
        { type: "text", value: ` (${lintSystem})` },
      ],
    },
    {
      type: "list",
      ordered: false,
      children: lintItems.map(
        (item: string): ListItem => ({
          type: "listItem",
          children: [
            { type: "paragraph", children: [{ type: "text", value: item }] },
          ],
        })
      ),
    },
  ];

  return segment;
};
