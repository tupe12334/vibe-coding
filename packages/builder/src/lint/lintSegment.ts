import type { ListItem, RootContent } from "mdast";
import { eslintRules } from "./eslint";

const lintRulesMap: Record<string, readonly string[]> = {
  eslint: eslintRules,
};

export const lintSegment = async (
  lintSystem: string
): Promise<RootContent[]> => {
  const lintItems = lintRulesMap[lintSystem] ?? [];

  const segment: RootContent[] = [
    {
      type: "heading",
      depth: 2,
      children: [
        { type: "text", value: "Lint system" },
        { type: "text", value: ` (${lintSystem})` },
      ],
    },
    {
      type: "list",
      ordered: false,
      children: lintItems.map(
        (item: string): ListItem => ({
          type: "listItem",
          children: [
            { type: "paragraph", children: [{ type: "text", value: item }] },
          ],
        })
      ),
    },
  ];

  return segment;
};
