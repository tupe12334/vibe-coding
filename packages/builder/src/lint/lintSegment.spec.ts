import { describe, expect, test } from "vitest";
import { lintSegment } from "./lintSegment";

describe("lintSegment", () => {
  test("returns lint system guidelines", async () => {
    const segment = await lintSegment("eslint");
    expect(segment).toMatchSnapshot();
  });
});
