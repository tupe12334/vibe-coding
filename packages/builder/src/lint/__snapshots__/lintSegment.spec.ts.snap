// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`lintSegment > returns lint system guidelines 1`] = `
[
  {
    "children": [
      {
        "type": "text",
        "value": "Lint system",
      },
      {
        "type": "text",
        "value": " (eslint)",
      },
    ],
    "depth": 2,
    "type": "heading",
  },
  {
    "children": [
      {
        "children": [
          {
            "children": [
              {
                "type": "text",
                "value": "Use ESLint to enforce code quality and style guidelines.",
              },
            ],
            "type": "paragraph",
          },
        ],
        "type": "listItem",
      },
      {
        "children": [
          {
            "children": [
              {
                "type": "text",
                "value": "Fix lint issues before committing code.",
              },
            ],
            "type": "paragraph",
          },
        ],
        "type": "listItem",
      },
    ],
    "ordered": false,
    "type": "list",
  },
]
`;
